src/storage/file_io.c

```diff
@@ -11873,3 +11875,136 @@ fileio_is_formatted_page (THREAD_ENTRY * thread_p, const char *io_page)
+ *
+ * NOTE: This function receives the path information for a LOB directory
+ *       and recursively deletes the corresponding physical directory at the OS level.
+ *       The input parameter 'path_key' can take two forms:
+ *       1. keyword: If a keyword is provided (e.g., "ces" or "hfid"..),
+ *          all directories in the LOB directory that include the keyword will be deleted.
+ *       2. lob directory name: If a directory name under the LOB directory is provided,
+ *          the function will traverse and delete that specific directory recursively.
+ */
+int
+fileio_lob_remove_dir (char *path_key)
+{
+#if defined(SERVER_MODE) || defined(SA_MODE)
+  DIR *dir_p;
+  struct dirent *dir_entry;
+  struct stat statbuf;
+  char base_dir[PATH_MAX], full_path[PATH_MAX], re_path[PATH_MAX];
+  char *pos, *keyword;
+  int result, sub_result;
+  result = sub_result = 0;
+  int error = NO_ERROR;
+
+  if (stat (es_base_dir, &statbuf) != 0 || !S_ISDIR (statbuf.st_mode))
+    {
+      error = ER_ES_NO_LOB_PATH;
+      return error;
+    }
+  dir_p = opendir (es_base_dir);
+  if (dir_p == NULL)
+    {
+      error = ER_ES_NO_LOB_PATH;
+      return error;
+    }
+  closedir (dir_p);
+
+  pos = strrchr (path_key, '/');
+  if (pos != NULL)
+    {
+      keyword = NULL;
+      snprintf (base_dir, (strlen (es_base_dir) + 1 + strlen (path_key) + 1), "%s/%s", es_base_dir, path_key);
+
+      if (base_dir[strlen (base_dir)] == '/')
+	{
+	  base_dir[strlen (base_dir)] = '\0';
+	}
+    }
+  else
+    {
+      snprintf (base_dir, (strlen (es_base_dir) + 1), "%s", es_base_dir);
+      keyword = path_key;
+    }
+
+  dir_p = opendir (base_dir);
+  if (dir_p == NULL)
+    {
+      error = ER_ES_INVALID_PATH;
+      return error;
+    }
+
+  while ((dir_entry = readdir (dir_p)) != NULL && result == 0)
+    {
+      if (strcmp (dir_entry->d_name, ".") == 0 || strcmp (dir_entry->d_name, "..") == 0)
+	{
+	  continue;
+	}
+
+      snprintf (full_path, (strlen (base_dir) + 1 + strlen (dir_entry->d_name) + 1), "%s/%s", base_dir,
+		dir_entry->d_name);
+
+      if (keyword != NULL)
+	{
+	  if (strncmp (dir_entry->d_name, keyword, strlen (keyword)) == 0)
+	    {
+	      if (stat (full_path, &statbuf) != 0)
+		{
+		  continue;
+		}
+
+	      if (S_ISDIR (statbuf.st_mode))
+		{
+		  snprintf (re_path, (strlen (dir_entry->d_name) + 2), "%s/", dir_entry->d_name);
+		  sub_result = fileio_lob_remove_dir (re_path);
+
+		  rmdir (full_path);
+		}
+	      else
+		{
+		  /* This case should never happen. */
+		}
+	    }
+	}
+      else
+	{
+	  if (stat (full_path, &statbuf) != 0)
+	    {
+	      continue;
+	    }
+
+	  if (S_ISDIR (statbuf.st_mode))
+	    {
+	      snprintf (re_path, (strlen (path_key) + 1 + strlen (dir_entry->d_name) + 1), "%s/%s/", path_key,
+			dir_entry->d_name);
+
+	      sub_result = fileio_lob_remove_dir (re_path);
+
+	      rmdir (full_path);
+	    }
+	  else
+	    {
+	      sub_result = unlink (full_path);
+	    }
+	}
+
+      result = sub_result;
```

Comment:
1. sub_result는 필요하지 않을 것 같습니다. 모두 result로 받으면 됩니다.
2. result != 0(에러)인 경우 error가 설정되지 않기 때문에 NO_ERROR를 리턴합니다. 추가적으로 에러가 발생하는 경우 er_set()이 기본적으로 필요합니다. 설정하지 않은 이유가 있으시면 회신주세요.

Reply:
```suggestion
      error = result;
```
함수 종료 부분에 위 코드 추가하겠습니다.

er_set()는 활용법을 알지 못해서 넣지 않았습니다.